refactor(cli): Update the cloud config command

owenrumney opened the pull request:
## Description

- update cloud config command to use set and get to complement the show
  and edit commands.
  - `trivy cloud config set`
  - `trivy cloud config get`
  - `trivy cloud config list`
  - `trivy cloud config edit`

`edit` command still opens with $EDITOR env var with fallback to platform
specific editor.

`list` replaces show and now has the dot notation names for the config
settings.


If an unknown setting is `set` an error is returned, likewise for the `get`

```bash
trivy cloud config set server.scanning.foo off

FATAL	Fatal error	failed to set attribute "server.scanning.foo": field "foo" not found in config
```

Documentation is updated to reflect the change


## Related issues
- Close #9677

## Output Example

```bash
trivy cloud config set server.scanning.enabled on

trivy cloud config set server.scanning.upload-results true

trivy cloud config set server.url https://scan.trivy.cloud
```


```bash
trivy cloud config list

Trivy Cloud Configuration
-------------------------
Filepath:  /Users/owen/.trivy/trivy-cloud.yaml
Logged In: Yes

api.url                           https://api.trivy.dev
server.url                        https://scan.trivy.dev
server.scanning.enabled           Enabled
server.scanning.upload-results    Enabled
server.scanning.secret-config     Disabled
server.scanning.misconfig-config  Disabled
```

The actual config is still stored in yaml but is now structured

```yaml
---
api:
    url: https://api.trivy.dev
server:
    url: https://scan.trivy.dev
    scanning:
        enabled: true
        upload-results: true
        secret-config: false
        misconfig-config: false
```

## Checklist
- [x] I've read the [guidelines for contributing](https://trivy.dev/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://trivy.dev/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)

nikpivkin commented on pkg/cloud/config.go:
```diff
@@ -197,25 +200,77 @@ func ShowConfig() error {
+	for _, field := range fields {
+		if len(field.path) > maxKeyLen {
+			maxKeyLen = len(field.path)
+		}
```
```suggestion
		maxKeyLen = max(maxKeyLen, len(field.path))
```

nikpivkin commented on pkg/cloud/config.go:
```diff
@@ -197,25 +200,77 @@ func ShowConfig() error {
+	}
+
+	for _, field := range fields {
+		fmt.Printf("%-*s  %s\n", maxKeyLen, field.path, formatValue(field.value))
```
Interesting, I didn't know about this formatting.

nikpivkin commented on pkg/cloud/config_modify.go:
```diff
@@ -0,0 +1,130 @@
+		case reflect.Float32, reflect.Float64:
+			return reflect.Value{}, xerrors.New("float conversion from string not yet implemented")
+		}
+	}
```
You could use YAML-based conversion here — it automatically handles all types (bool, int, float) and removes the need for the manual switch/case. Performance will be slightly lower, but it’s not critical for working with the config. Wdyt?

```suggestion
func convertToType(value any, targetType reflect.Type) (reflect.Value, error) {
	val := reflect.ValueOf(value)
	if val.Type().AssignableTo(targetType) {
		return val, nil
	}

	targetPtr := reflect.New(targetType) // *T
	targetInterface := targetPtr.Interface()

	data, err := yaml.Marshal(value)
	if err != nil {
		return reflect.Value{}, xerrors.Errorf("failed to marshal value: %w", err)
	}

	if err := yaml.Unmarshal(data, targetInterface); err != nil {
		return reflect.Value{}, xerrors.Errorf("failed to decode into %v: %w", targetType, err)
	}

	return targetPtr.Elem(), nil
}
```

nikpivkin commented:
@owenrumney Thanks! That makes sense. 

I have another question about the configuration. Should we consider adding a `config reset` or `config unset` command?
It would allow users to quickly restore default settings or remove specific attributes.

For reference, similar patterns exist in other CLIs:
- `kubectl config unset <key>`
- `git config unset <key>`
- `npm config delete <key>`

owenrumney commented:
Added the `unset` command to `cloud config` and added some examples to the `get/set/unset`

```
$ trivy cloud config set --help
Set the Trivy Cloud platform configuration

Usage:
  trivy cloud config set [setting] [value] [flags]

Examples:
  $ trivy cloud config set server.scanning.enabled true
  $ trivy cloud config set server.scanning.upload-results false

Flags:
  -h, --help   help for set

Global Flags:
      --cache-dir string          cache directory (default "/Users/owen/Library/Caches/trivy")
  -c, --config string             config path (default "trivy.yaml")
  -d, --debug                     debug mode
      --generate-default-config   write the default config to trivy-default.yaml
      --insecure                  allow insecure server connections
  -q, --quiet                     suppress progress bar and log output
      --timeout duration          timeout (default 5m0s)
  -v, --version                   show version
```

DmitriyLewen commented on pkg/commands/cloud/run.go:
```diff
@@ -95,10 +95,26 @@ func CheckTrivyCloudStatus(cmd *cobra.Command) error {
 }
 
-func ShowConfig() error {
+func ListConfig() error {
```
```bash
➜ cat /Users/dmitriy/.trivy/trivy-cloud.yaml

cat: /Users/dmitriy/.trivy/trivy-cloud.yaml: No such file or directory

➜  ./trivy cloud config list                 

Trivy Cloud Configuration
-------------------------
Filepath:  /Users/dmitriy/.trivy/trivy-cloud.yaml
Logged In: No

api.url                           https://api.trivy.dev
server.url                        https://scan.trivy.dev
server.scanning.enabled           Enabled
server.scanning.upload-results    Enabled
server.scanning.secret-config     Enabled
server.scanning.misconfig-config  Enabled

➜ ./trivy cloud config get server.url
https://scan.trivy.dev
```

Do we actually want to show default config if the file doesn’t exist?
That could be confusing.
For example, a user hasn’t logged in yet but sees server.scanning.upload-results    Enabled.
However, we won’t send a report because there’s no token.

owenrumney replied:
I added the `Logged In` to the list configuration to show that they weren't logged in, but yeah, maybe you're right that certainly the `server.scanning.enabled` should be false, probably the others too. Will take a look.
